pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -254,6 +280,9 @@ func (pl *PodTopologySpread) calPreFilterState(ctx context.Context, pod *v1.Pod)
 
 	for _, tpCounts := range tpCountsByNode {
 		for tp, count := range tpCounts {
+			if _, ok := s.TpPairToMatchNum[tp]; !ok {
+				s.TpKeyToDomainsNum[tp.key]++
+			}
 			s.TpPairToMatchNum[tp] += count
```

Comment:
alternate implementation, to be done as a separate loop after this one

```golang
for tp := range s.TpPairToMatcNum {
  s.TpKeyToDomainsNum[tp.key]++
}
```

This is potentially faster, because map accesses are not cheap. It also easier to guard the loop by checking the feature flag :)